lib/withOnyx.js

```diff
@@ -88,6 +101,14 @@ export default function (mapOnyxToState) {
              * @param {*} val
              */
             setWithOnyxState(statePropertyName, val) {
+                // This method gets called by Onyx.connect no matter if the cached
+                // data is already there or not. Thus we have to check whether
```

Comment:
NAB, but maybe we should not call this at all in the case the cached data is there. This makes more sense to me vs. letting it get called but "cancelling" it. I also think the code we have here works no NAB.

Reply:
the condition is paired with the loading state, which is something the `withOnyx` is aware about, while `Onyx.js` isn't really. Hence I think it makes sense to leave the condition in this place 🤔